src/codegen/generators/pou_generator.rs

```diff
@@ -320,27 +317,34 @@ impl<'ink, 'cg> PouGenerator<'ink, 'cg> {
     fn create_llvm_function_type(
         &self,
         parameters: Vec<BasicMetadataTypeEnum<'ink>>,
-        is_var_args: bool,
+        variadic: Option<&'cg VarArgs>,
         return_type: Option<BasicTypeEnum<'ink>>,
     ) -> Result<FunctionType<'ink>, Diagnostic> {
-        let params = parameters.as_slice();
+        //Sized variadic is not considered a variadic function, but receives 2 extra parameters, size and a pointer
+        let is_var_args = variadic.map(|it| !it.is_sized()).unwrap_or_default();
+        let sized_variadics = self.get_sized_variadic_params(variadic);
```

Comment:
can we rename sized_variadics to size_and_pointer?
you explain above that this is no real variadic